packages/flutter_test/test/accessibility_test.dart

```diff
@@ -23,6 +23,59 @@ void main() {
       handle.dispose();
     });
 
+    testWidgets('Multiple text with same label', (WidgetTester tester) async {
+      final SemanticsHandle handle = tester.ensureSemantics();
+      await tester.pumpWidget(
+        _boilerplate(
+          Column(
+            children: const <Widget>[
+              Text(
+                'this is a test',
+                style: TextStyle(fontSize: 14.0, color: Colors.black),
+              ),
+              Text(
+                'this is a test',
+                style: TextStyle(fontSize: 14.0, color: Colors.black),
+              ),
+            ],
+          ),
+        ),
+      );
+      await expectLater(tester, meetsGuideline(textContrastGuideline));
+      handle.dispose();
+    });
+
+    testWidgets(
+      'Multiple text with same label but Nodes excluded from '
+      'semantic tree have failing contrast should pass a11y guideline ',
+      (WidgetTester tester) async {
+        final SemanticsHandle handle = tester.ensureSemantics();
+        await tester.pumpWidget(
+          _boilerplate(
+            Column(
+              children: const <Widget>[
+                Text(
+                  'this is a test',
+                  style: TextStyle(fontSize: 14.0, color: Colors.black),
+                ),
+                Text(
+                  'this is a test',
+                  style: TextStyle(fontSize: 14.0, color: Colors.black),
+                ),
+                ExcludeSemantics(
```

Comment:
If I run this test without your changes (and with an added # to avoid the duplicate label check) then it still passes with / without the ExcludeSemantics

Please verify the output of this test

Reply:
The third text which is wrapped in ExcludeSemantics has text color white while the first 2 have text color black. 

The third text doesn't pass the contrast requirement (White text on white background). It only passes when ExcludeSemantics is added on the test.

Reply:
If I run this locally on ToT, remove the exclude semantics, and change the text so that all 3 labels are distinct - the test still passes

Reply:
Given that the test framework is broken (Ref https://github.com/flutter/flutter/pull/109811/files), I don't think it makes sense to block other updates to the infra unless there is a repro that proves that the current fix is not valid.

Reply:
I agree its unfortunate, but if we're not able to ensure that this change you've posted works as expected I don't see how it could possibly be safe to land it. 

> unless there is a repro that proves that the current fix is not valid.

I think that is a bit backwards, no? The feature needs to add some tests that provide some reasonable confidence that it works as expected.  From the PR above, we can see there is an issue with that particular kind of test that needs to be fixed. 

FYI @Hixie

Reply:
At TOT the current test is broken. The test fails just because there are 2 nodes with same semantic label. With the current setup there is nothing the developer can do to fix the test other than changing the UI to test (which may not be feasible at most times) 

With the proposed change, the only mistake it would make is telling the developer the wrong semantic node number for which the contrast is failing. For example if there are 3 texts (`T1`, `T2`, `T3`) named `ABC`, and `T3` is actually failing contrast, the test would say `T1, T2, T3` are failing. But given all texts are same content, the developer should be able to fix it. 

You might say if I update the language of the failure to say *at least* one of the `Text with content ABC` is failing contrast test, it would be technically correct in all sense.

Reply:
On the Flutter team we have a saying, "embrace the yak shave", which applies here. Before landing this PR, the underlying issues that were raised here should be fixed.

Reply:
I added an updated test #110217 (No need to merge #110217) 

The original test was passing even though it should have failed was because of the offset in Element paint checker which adds a 4dp offset on all sides which causes the contrast to pick colors from neighboring texts.  

Also updated the check itself to ignore all elements where Node's rect intersect with the Element's rect so that we ignore all other Element's which are not related to the semantic node.

Reply:
FYI @chunhtai , another example that the custom font would help

Reply:
The custom font is tracked in https://github.com/flutter/flutter/issues/103235

Reply:
I think the issue in #103235 can be fixed by removing the offset in the test. Not really sure the reason behind adding a 4dp offset around the paint bounds in all direction. That is just picking up stray pixels from other widgets.

Reply:
Because we use Ahem which is solid rectangles, there is a risk we can't pick up the background color.

Reply:
Question: why text contrast fails in exclude semantics should pass the test? Text contrast is pixel based, has nothing to do with semantics. Even if assistive technology users can't interact with the widget, regular user can still see the widget on the screen. The text contrast requirement should still apply.

Reply:
There are many reasons that something can be excluded from semantics, for example being partially covered or de-emphasized in the UI that would also make contrast computation irrelevant